service/src/main/kotlin/fi/espoo/evaka/reports/FuturePreschoolersReport.kt

```diff
@@ -0,0 +1,216 @@
+    fun getPreschoolGroups(
+        db: Database,
+        user: AuthenticatedUser,
+        clock: EvakaClock,
+        @RequestParam municipal: Boolean
+    ): List<PreschoolGroupsReportRow> {
+        return db.connect { dbc ->
+                dbc.read {
+                    accessControl.requirePermissionFor(
+                        it,
+                        user,
+                        clock,
+                        Action.Global.READ_FUTURE_PRESCHOOLERS
+                    )
+                    it.setStatementTimeout(REPORT_STATEMENT_TIMEOUT)
+                    it.getPreschoolGroupsRows(clock.today(), municipal)
+                }
+            }
+            .also { Audit.FuturePreschoolers.log(meta = mapOf("count" to it.size)) }
+    }
+}
+
+const val preschoolSelectionAge = 5
+
+fun Database.Read.getFuturePreschoolerRows(today: LocalDate): List<FuturePreschoolersReportRow> =
+    createQuery<DatabaseTable> {
+            sql(
+                """
+SELECT p.id, 
+    p.social_security_number AS child_ssn,
+    p.last_name AS child_last_name,
+    p.first_name AS child_first_name,
+    p.street_address AS child_address,
+    p.postal_code AS child_postal_code,
+    upper(p.post_office) AS child_post_office,
+    d.name AS unit_name,
+    d.street_address AS unit_address,
+    d.postal_code AS unit_postal_code,
+    upper(d.post_office) AS unit_post_office,
+    NULL as unit_area, -- cannot be obtained from current evaka dataset, care area is not precise enough
+    g1.last_name AS guardian_1_last_name,
+    g1.first_name AS guardian_1_first_name,
+    g1.street_address AS guardian_1_address,
+    g1.postal_code AS guardian_1_postal_code,
+    upper(g1.post_office) AS guardian_1_post_office, 
+    g1.phone AS guardian_1_phone,
+    g1.email AS guardian_1_email,
+    g2.last_name AS guardian_2_last_name, 
+    g2.first_name AS guardian_2_first_name, 
+    g2.street_address AS guardian_2_address,
+    g2.postal_code AS guardian_2_postal_code,
+    upper(g2.post_office) AS guardian_2_post_office, 
+    g2.phone AS guardian_2_phone,
+    g2.email AS guardian_2_email,
+    CASE WHEN sn.shift_care = 'FULL'::shift_care_type OR sn.shift_care = 'INTERMITTENT'::shift_care_type THEN true ELSE false END AS shift_care,
+    NULL AS language_emphasis_group, -- cannot be obtained from current evaka dataset
+    CASE WHEN sno.name_fi like 'Kaksivuotinen%' THEN true ELSE false END AS two_year_preschool
+FROM person p
+JOIN placement pl ON pl.child_id = p.id
+JOIN daycare d ON d.id = pl.unit_id
+LEFT JOIN (
+    SELECT id, last_name, first_name, street_address, postal_code, post_office, phone, email
+    FROM person
+) g1 ON g1.id = (
+    SELECT guardian_id AS id FROM guardian WHERE child_id = p.id
+    UNION
+    SELECT parent_id AS id FROM foster_parent WHERE child_id = p.id AND valid_during @> :today
+    EXCEPT 
+    SELECT guardian_id AS id FROM guardian_blocklist where child_id = p.id
```

Comment:
`guardian_blocklist` is only used to prevent automatically adding VTJ guardians to the `guardian` table, so this `EXCEPT` is not needed.